library/net.c

```diff
@@ -751,6 +751,9 @@ int dnet_process_recv(struct dnet_backend_io *backend, struct dnet_net_state *st
 
 void dnet_state_remove_nolock(struct dnet_net_state *st)
 {
+	if (st->__join_state == DNET_JOIN)
```

Comment:
There was a single list previously. For client node it should contain the same nodes, since client can only connect to server nodes. While server node can connect to server nodes only, it can also accept client nodes. But only client is interested in number of connections and only interested in server connections.

Actually, I do not think this is still needed, since number of connections is very high these days. It was introduced when number of connections was around ten and we checked whether proxy is connected to at least half of them. With mastermind/backrunner management it is not needed.

Please talk about this number with colleagues, maybe we can remove it altogether. We do not use it in reverbrain proxies at all.